pdns/mastercommunicator.cc

```diff
@@ -248,7 +251,10 @@ void CommunicatorClass::sendNotification(int sock, const DNSName& domain, const
     trc.d_fudge = 300;
     trc.d_origID=ntohs(id);
     trc.d_eRcode=0;
-    B64Decode(tsigsecret64, tsigsecret);
+    if (!B64Decode(tsigsecret64, tsigsecret)) {
```

Comment:
Isn't that the other way around? It looks like `B64Decode` returns `0` when everything is fine, although our unit tests are not checking that at all..

Reply:
you are right, looking at the code for B64Decode I wonder why we don't just return a bool.....